mqtt/src/main/scala/akka/stream/alpakka/mqtt/MqttFlowStage.scala

```diff
@@ -90,7 +92,7 @@ final class MqttFlowStage(sourceSettings: MqttSourceSettings,
       }
 
       private val onPublished = getAsyncCallback[Try[IMqttToken]] {
-        case Success(_) => pull(in)
+        case Success(_) => if (!hasBeenPulled(in)) pull(in)
```

Comment:
An additional check as I noticed a scenario where we had pulled already given auto-reconnect.